fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/PlanWithStoredFields.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * PlanWithStoredFields.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.plan;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.metadata.expressions.KeyExpression;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+/**
+ * A plan that has some, but not all, stored fields.
+ */
+@API(API.Status.INTERNAL)
+public interface PlanWithStoredFields {
+    void getStoredFields(@Nonnull List<KeyExpression> keyFields, @Nonnull List<KeyExpression> nonStoredFields);
```

Comment:
Can you please add a javadoc on how this method needs to be implemented? Its name is confusing as it's not really returning anything. I am assuming the implementor is partitioning its fields.

Reply:
Added.